lib/liveman/survey/surveys.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Liveman.Survey.Surveys do
+  use Ecto.Schema
+
+  alias Liveman.Survey.Schemas.Question
+  alias Liveman.Survey.Schemas.Survey
+
+  @json_file "priv/repo/data/surveys.json"
+
+  def list_surveys do
+    survey_list_json = get_json()
+
+    Enum.map(survey_list_json, fn survey_json ->
+      survey = build_survey(survey_json)
+      questions = build_question(survey_json["questions"], survey)
+
+      survey
+      |> Ecto.Changeset.change()
+      |> Ecto.Changeset.put_embed(:questions, questions)
+      |> Ecto.Changeset.apply_changes()
```

Comment:
I wonder if we can use `cast_embed` instead. I think it would put the embed and apply the changes at the same time 🤔

Reply:
Checked more. In this feature, we actually directly build the survey embedded_schema including associated embedded_schema (`questions`) without `cast_embed` and then pass it to the JSON serializer. It works.

I still added `cast_embed` in case we need to update the survey schema with the association, it would be useful at that time.